apps/wallet/src/ui/app/hooks/deepbook/useGetEstimate.ts

```diff
@@ -0,0 +1,287 @@
+			undefined,
+			undefined,
+			address,
+			txb,
+		);
+	} else {
+		const primaryCoinInput = txb.object(quoteCoins[0].coinObjectId);
+		const restCoins = quoteCoins.slice(1);
+
+		if (restCoins.length) {
+			txb.mergeCoins(
+				primaryCoinInput,
+				restCoins.map((coin) => txb.object(coin.coinObjectId)),
+			);
+		}
+
+		const walletFee = new BigNumber(quoteBalance)
+			.times(WALLET_FEES_PERCENTAGE / 100)
+			.integerValue(BigNumber.ROUND_DOWN)
+			.toString();
+
+		balanceToSwap = new BigNumber(quoteBalance).minus(walletFee).toString();
+
+		const [swapCoin, walletCoin] = txb.splitCoins(primaryCoinInput, [balanceToSwap, walletFee]);
+
+		txnResult = await deepBookClient.swapExactQuoteForBase(
+			poolId,
+			swapCoin,
+			BigInt(balanceToSwap),
+			address,
+			undefined,
+			txb,
+		);
+
+		walletFeeCoin = walletCoin;
+	}
+
+	if (!accountCapId) {
+		txnResult.transferObjects([accountCap], address);
+	}
+
+	if (walletFeeCoin) txnResult.transferObjects([walletFeeCoin], walletFeeAddress);
+
+	return txnResult;
+}
+
+export function useGetEstimate({
+	accountCapId,
+	signer,
+	coinType,
+	poolId,
+	baseBalance,
+	quoteBalance,
+	isAsk,
+}: {
+	accountCapId: string;
+	signer: WalletSigner | null;
+	coinType: string;
+	poolId: string;
+	baseBalance: string;
+	quoteBalance: string;
+	isAsk: boolean;
+}) {
+	const walletFeeAddress = useFeatureValue(FEATURES.WALLET_FEE_ADDRESS, DEFAULT_WALLET_FEE_ADDRESS);
+	const queryClient = useQueryClient();
+	const suiClient = useSuiClient();
+	const activeAccount = useActiveAccount();
+	const activeAddress = activeAccount?.address;
+	const deepBookClient = useDeepBookContext().client;
+
+	const { data } = useGetObject(poolId);
+	const objectFields =
+		data?.data?.content?.dataType === 'moveObject' ? data?.data?.content?.fields : null;
+
+	const lotSize = (objectFields as Record<string, string>)?.lot_size;
+
+	return useQuery({
+		// eslint-disable-next-line @tanstack/query/exhaustive-deps
+		queryKey: [
+			DEEPBOOK_KEY,
+			'get-estimate',
+			poolId,
+			accountCapId,
+			coinType,
+			activeAddress,
+			baseBalance,
+			quoteBalance,
+			isAsk,
+			lotSize,
+		],
+		queryFn: async () => {
+			const [baseCoins, quoteCoins] = await Promise.all([
+				getCoinsByBalance({
+					coinType,
+					balance: baseBalance,
+					suiClient,
+					address: activeAddress!,
+				}),
+				getCoinsByBalance({
+					coinType,
+					balance: quoteBalance,
+					suiClient,
+					address: activeAddress!,
+				}),
+			]);
+
+			if ((isAsk && !baseCoins.length) || (!isAsk && !quoteCoins.length)) {
+				throw new Error('No coins found in balance');
+			}
+
+			const txn = await getPlaceMarketOrderTxn({
+				deepBookClient,
+				poolId,
+				accountCapId,
+				address: activeAddress!,
+				isAsk,
+				lotSize: Number(lotSize),
+				baseCoins,
+				quoteCoins,
+				baseBalance,
+				quoteBalance,
+				walletFeeAddress,
+			});
+
+			if (!accountCapId) {
+				await queryClient.invalidateQueries(['get-owned-objects']);
```

Comment:
with react-query v5 that just landed you'll need to change this to:

```suggestion
				await queryClient.invalidateQueries({ queryKey: ['get-owned-objects'] });
```

Reply:
updated